deployment/data-streams/changeset/jd_register_nodes.go

```
@@ -3,37 +3,90 @@ package changeset
+		return "oracle"
+	case NodeTypeBootstrap:
+		return "bootstrap"
+	default:
+		return "unknown"
+	}
+}
+
 type RegisterNodesInput struct {
 	EnvLabel    string
 	ProductName string
-	DONs        DONConfigMap
+	// Will be deleted after migration to DONConfigMap
+	DONs     DONConfigMap `json:"dons,omitempty"`
+	DONsList []DONConfig  `json:"dons_list,omitempty"`
 }
 
 type DONConfigMap map[string]DONConfig
 
 type DONConfig struct {
+	ID                 int       `json:"id"`
+	ChainSelector      string    `json:"chainSelector"`
 	Name               string    `json:"name"`
 	ChannelConfigStore string    `json:"channelConfigStore"`
 	Verifier           string    `json:"verifier"`
 	Configurator       string    `json:"configurator"`
 	Nodes              []NodeCfg `json:"nodes"`
+	BootstrapNodes     []NodeCfg `json:"bootstrapNodes"`
 }
 
 type NodeCfg struct {
 	Name        string `json:"name"`
 	CSAKey      string `json:"csa_key"`
-	IsBootstrap bool   `json:"isBootstrap"`
+	IsBootstrap bool   `json:"isBootstrap,omitempty"`
+}
+
+func validateNodeSlice(nodes []NodeCfg, nodeType string, donIndex int) error {
+	for _, node := range nodes {
+		if node.Name == "" {
+			return fmt.Errorf("DON[%d] has %s node with empty Name", donIndex, nodeType)
+		}
+		if node.CSAKey == "" {
+			return fmt.Errorf("DON[%d] %s node %s has empty CSAKey", donIndex, nodeType, node.Name)
+		}
+	}
+	return nil
+}
+
+func registerNodesForDON(e deployment.Environment, nodes []NodeCfg, baseLabels []*ptypes.Label, nodeType NodeType) {
+	ntStr := nodeType.String()
+	for _, node := range nodes {
+		labels := append([]*ptypes.Label(nil), baseLabels...)
+		labels = append(labels, &ptypes.Label{
+			Key:   "nodeType",
+			Value: &ntStr,
+		})
+		nodeID, err := e.Offchain.RegisterNode(e.GetContext(), &nodev1.RegisterNodeRequest{
+			Name:      node.Name,
+			PublicKey: node.CSAKey,
+			Labels:    labels,
+		})
+		if err != nil {
+			e.Logger.Errorw("failed to register node", "node", node.Name, "error", err)
```

should we return the error here? or is it intentional to continue if not registered succesfully?

---

Is Intentional, but I left a comment so we can revisit this if necessary in the future.